numpy/core/src/multiarray/sequence.c

```diff
@@ -27,90 +27,6 @@ array_any_nonzero(PyArrayObject *mp);
    we fill it in here so that PySequence_XXXX calls work as expected
 */
 
-
-static PyObject *
-array_slice(PyArrayObject *self, Py_ssize_t ilow, Py_ssize_t ihigh)
-{
-    PyArrayObject *ret;
-    PyArray_Descr *dtype;
-    Py_ssize_t dim0;
-    char *data;
-    npy_intp shape[NPY_MAXDIMS];
-
-    if (PyArray_NDIM(self) == 0) {
-        PyErr_SetString(PyExc_ValueError, "cannot slice a 0-d array");
-        return NULL;
-    }
-
-    dim0 = PyArray_DIM(self, 0);
-    if (ilow < 0) {
-        ilow = 0;
-    }
-    else if (ilow > dim0) {
-        ilow = dim0;
-    }
-    if (ihigh < ilow) {
-        ihigh = ilow;
-    }
-    else if (ihigh > dim0) {
-        ihigh = dim0;
-    }
-
-    data = PyArray_DATA(self);
-    if (ilow < ihigh) {
-        data += ilow * PyArray_STRIDE(self, 0);
-    }
-
-    /* Same shape except dimension 0 */
-    shape[0] = ihigh - ilow;
-    memcpy(shape+1, PyArray_DIMS(self) + 1,
-                        (PyArray_NDIM(self)-1)*sizeof(npy_intp));
-
-    dtype = PyArray_DESCR(self);
-    Py_INCREF(dtype);
-    ret = (PyArrayObject *)PyArray_NewFromDescr(Py_TYPE(self), dtype,
-                             PyArray_NDIM(self), shape,
-                             PyArray_STRIDES(self), data,
-                             PyArray_FLAGS(self),
-                             (PyObject *)self);
-    if (ret == NULL) {
-        return NULL;
-    }
-    Py_INCREF(self);
-    if (PyArray_SetBaseObject(ret, (PyObject *)self) < 0) {
-        Py_DECREF(ret);
-        return NULL;
-    }
-    PyArray_UpdateFlags(ret, NPY_ARRAY_UPDATE_ALL);
```

Comment:
Hmmm, no idea why this should be needed here. Maybe it was just being very careful, or a git blame might help to see when the line was added if it was. The array creation should set all the flags correctly already, except maybe writable, but that is correct.